### PR TITLE
feat: proxy-mirror support grpc and grpcs

### LINKED ISSUE #7437: feat: proxy-mirror support grpc protocol
### Description

As a user, I want the proxy-mirror plugin can also support grpc protocol, so that I can use it to copy grpc traffic to a special upstream and store it for replaying

### PR DESCRIPTION
**Pull request by Sn0rt**
### Description

Fixes #7437 #9245

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/cli/ngx_tpl.lua [hunk 1]
```diff
@@ -738,6 +740,7 @@ http {
             mirror          /proxy_mirror;
             {% end %}
 
+
```
Comment (kingluo) [inline apisix/cli/ngx_tpl.lua:743]: remove this duplicated empty line.

File: apisix/cli/ngx_tpl.lua [hunk 2]
```diff
@@ -833,6 +840,32 @@ http {
...
+            {% if not use_apisix_base then %}
+            if ($upstream_mirror_uri = "") {
+                return 200;
+            }
+            {% end %}
+
+
+            {% if proxy_mirror_timeouts then %}
+                {% if proxy_mirror_timeouts.connect then %}
+            proxy_connect_timeout {* proxy_mirror_timeouts.connect *};
```
Comment (kingluo) [inline apisix/cli/ngx_tpl.lua:857]: Should use `grpc_*_timeout`
http://nginx.org/en/docs/http/ngx_http_grpc_module.html#grpc_connect_timeout

Comment (Sn0rt) [reply apisix/cli/ngx_tpl.lua]: updated.

File: apisix/cli/ngx_tpl.lua (hunk 1 above)
Comment (Sn0rt) [reply apisix/cli/ngx_tpl.lua]: done

**Pull request closed without merging**